frontend/src/components/lineage/LineageNode.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import {
+  GraphElement,
+  isNode,
+  Node,
+  observer,
+  WithSelectionProps,
+  TaskNode,
+  RunStatus,
+  useHover,
+  ScaleDetailsLevel,
+  useAnchor,
+  AnchorEnd,
+  TaskNodeSourceAnchor,
+  TaskNodeTargetAnchor,
+} from '@patternfly/react-topology';
+import { DatabaseIcon, CubeIcon, CodeBranchIcon, BuildIcon } from '@patternfly/react-icons';
+import {
+  chart_color_blue_200 as chartColorBlue,
+  chart_color_green_200 as chartColorGreen,
+  chart_color_purple_200 as chartColorPurple,
+  chart_color_black_500 as chartColorBlack,
+} from '@patternfly/react-tokens';
+import { useEdgeHighlighting } from './edge/edgeStateUtils';
+
+// Define the types of lineage entities
+export type LineageEntityType =
+  | 'entity'
+  | 'batch_data_source'
+  | 'push_data_source'
+  | 'request_data_source'
+  | 'batch_feature_view'
+  | 'on_demand_feature_view'
+  | 'stream_feature_view'
+  | 'feature_service';
+
+export interface LineageNodeData {
+  label: string;
+  entityType: LineageEntityType;
+  features?: number; // For feature views
+  description?: string;
+  truncateLength?: number; // Control text truncation (default: 30 for lineage)
+}
+
+type LineageNodeProps = {
+  element: GraphElement;
+} & WithSelectionProps;
+
+// Icon mapping for different entity types
+const getEntityIcon = (entityType: LineageEntityType, selected = false) => {
+  const iconColor = selected ? '#ffffff' : undefined;
+
+  switch (entityType) {
+    case 'entity':
+      return <CodeBranchIcon style={{ color: iconColor || chartColorBlack.value }} />;
+    case 'batch_data_source':
+    case 'push_data_source':
+    case 'request_data_source':
+      return <DatabaseIcon style={{ color: iconColor || chartColorBlue.value }} />;
+    case 'batch_feature_view':
+    case 'on_demand_feature_view':
+    case 'stream_feature_view':
+      return <BuildIcon style={{ color: iconColor || chartColorPurple.value }} />;
+    case 'feature_service':
+      return <BuildIcon style={{ color: iconColor || chartColorGreen.value }} />;
+    default:
+      return <CubeIcon style={{ color: iconColor || chartColorBlack.value }} />;
+  }
+};
+
+const LineageNodeInner: React.FC<{ element: Node } & WithSelectionProps> = observer(
+  ({ element, onSelect, selected }) => {
+    const data = element.getData();
+    const [hover, hoverRef] = useHover<SVGGElement>();
+    const detailsLevel = element.getGraph().getDetailsLevel();
+
+    // Get the current visualization state to check for highlighting
+    const { isConnectedToSelection } = useEdgeHighlighting(element.getId(), selected);
+
+    // Set up proper anchors for edge connections with better positioning
+    useAnchor(
+      React.useCallback(
+        (node: Node) => new TaskNodeSourceAnchor(node, ScaleDetailsLevel.high, 0, false),
```

Comment:
You might want to create your own Source Anchor to make the outgoing edges connect to the right edge of the node. You would have to compute the real node size to determine the source anchor location. It looks like the right badge and paddings would need to be computed and added to the location.

<img width="520" height="402" alt="image" src="https://github.com/user-attachments/assets/75f7db9d-bd00-4bc9-822d-f130b58eda92" />

Reply:
Tried your suggested approach, the Curve Edge that i have created is working with that. Closing this comment for now, will add next week, in my refinement.